### PR TITLE
fix: simplify file watching expression

### PR DESCRIPTION
**Pull request by eduardoboucas**
#### Summary

Follow-up to https://github.com/netlify/cli/pull/6018 addressing a couple of items from the code review.

### CODE REVIEW

File: src/lib/edge-functions/registry.mjs [hunk 1]
```diff
@@ -548,7 +548,7 @@ export class EdgeFunctionsRegistry {
    * @returns {Promise<void>}
    */
   async #setupWatcherForDirectory(directory) {
-    const ignored = [new RegExp(this.#servePath)]
+    const ignored = [`${this.#servePath}/**`]
```
Comment (lukasholzer) [inline src/lib/edge-functions/registry.mjs:551]: is this using unix paths on windows as well, I guess so but just checking.

File: tests/integration/commands/dev/edge-functions.test.ts [hunk 2]
```diff
@@ -30,9 +30,9 @@ const routes = [
   },
 ]
 
-describe('edge functions', () => {
+describe.skipIf(isWindows)('edge functions', () => {
```
Comment (lukasholzer) [inline tests/integration/commands/dev/edge-functions.test.ts:33]: why are they skipped on windows? shouldn't it work there as well?

Comment (eduardoboucas) [reply tests/integration/commands/dev/edge-functions.test.ts:33]: https://github.com/netlify/cli/blob/e08a47f2d362fa82a64237260b5bc517ca75a6b3/tests/integration/commands/dev/edge-functions.test.ts#L9-L10

File: src/lib/edge-functions/registry.mjs (hunk 1 above)
Comment (eduardoboucas) [reply src/lib/edge-functions/registry.mjs:551]: Yes: https://github.com/micromatch/micromatch#backslashes.

**Review by lukasholzer — APPROVED** [review]
But not blocking it :)

**Pull request merged**